UnitTest.java

```diff
@@ -0,0 +1,160 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Assertions;
+
+public class UnitTest {
+    private static Unit unit1;
+    private static Unit unit2;
+
+    @Test
+    @DisplayName("Test with constructor")
+    void test1() {
+        unit1 = new Unit("harcos", 15, 100, 1.5);
+        Assertions.assertEquals(unit1.getName(), "harcos", "Wrong name");
+        Assertions.assertEquals(unit1.getDMG(), 15, "Dmg is wrong");
+        Assertions.assertEquals(unit1.getHp(), 100, "Hp is wrong");
+        Assertions.assertEquals(unit1.getAs(), 1.5, "Attack speed is wrong");
+    }
+
+    @Test
+    @DisplayName("Test with hp")
+    void test2() {
+        unit1 = new Unit("harcos", 15, 100, 1.5);
+        Assertions.assertTrue(unit1.getHp() > 0, "hp can not be 0");
+    }
```

Comment:
ezt tovabbra sem ertem. itt miert nem a konkret hp-t kerdezed le. Illetve a battle-t teszteled parameteres tesztekkel, ahol ez nehezen atlathato. Ugyanez volt a masik csapatnal is, gondolom nem veletlen. De ponthogy itt lenne ertelme parameteres tesznek, mert itt tok egyertelmu. Ott egyet sem trivialis kod alapjan latni, hogy mire fog jutni egy battle, hat meg ha 10-et kell parhuzamosan. itt iszont tok egyertelmu, hogy amit megadsz a hp-nak, azt kell visszakapnod.